module/People/public/js/profile.js

```diff
@@ -64,19 +140,134 @@ Profile.prototype = {
 		$('#tdLastYear').html(creditsData.lastY);
 
 		container.show();
+	},
+
+	listTasks: function(url)
+	{
+		var that = this;
+		$.ajax({
+			url: url,
+			headers: {
+				'GOOGLE-JWT': sessionStorage.token
+			},
+			method: 'GET'
+		}).fail(function( jqXHR, textStatus ) {
+			var errorCode = jqXHR.status;
+			var redirectURL = window.location.href;
+			if(errorCode === 401){
+				sessionStorage.setItem('redirectURL', redirectURL);
+				window.location = '/';
+			}
+		}).done(that.onListTasksCompleted.bind(that));
+	},
+
+	onListTasksCompleted: function(data){
+		var container = this.createTaskMetricsTable(data);
+		if(data._links !== undefined && data._links["next"] !== undefined) {
+			var limit = this.getTasksPageSize() + this.getNextTasksPageSize();
+			container.append(
+				'<div class="text-center">' +
+						'<a rel="next" href="'+data._links["next"]["href"]+'?limit=' + limit + '" data-action="nextPage">More</a>' +
+				'</div>'
+			);
+		}
+	},
+
+	createTaskMetricsTable(data){
+		var tasks = data._embedded['ora:task'];
+		var container = $('#task-metrics');
+		var that = this;
+		var html = "";
+		html += "<table class=\"table table-hover\">";
+		html += "<thead>" +
+					"<tr>" +
+						"<th class=\"text-center\" style=\"width: 3em\"></th>" +
+						"<th class=\"text-left\">Subject</th>" +
+						"<th class=\"text-right\" style=\"width: 6em\">Credits</th>" +
+						"<th class=\"text-right\" style=\"width: 8em\">&Delta; shares" +
+					"</tr>" +
+				"</thead>";
+		html += "<tbody>";
+		if(!$.isEmptyObject(tasks)){
+			$.each(tasks, function(key, task) {
+				if(key.toUpperCase() == 'STATS'){
```

Comment:
Perché applichi la trasformazione in UpperCase? Stai analizzando una struttura dati nota, per cui sai se il nome di proprietà è maiuscolo o minuscolo